easybuild/framework/easyblock.py

```diff
@@ -2387,7 +2390,8 @@ def handle_iterate_opts(self):
                     self.log.debug("Iterating opt %s: %s", opt, self.iter_opts[opt])
 
             if self.iter_opts:
-                print_msg("starting iteration #%s ..." % self.iter_idx, log=self.log, silent=self.silent)
+                print_msg(f"starting iteration number {self.iter_idx + 1}/{self.iter_cnt} ...", log=self.log,
```

Comment:
I would drop the `number`:
```suggestion
                print_msg(f"starting iteration {self.iter_idx + 1}/{self.iter_cnt} ...", log=self.log,